triedb/pathdb/history_indexer.go

```diff
@@ -364,8 +371,8 @@ func (i *indexIniter) remain() uint64 {
 	default:
 		last, indexed := i.last.Load(), i.indexed.Load()
 		if last < indexed {
-			log.Error("Invalid state indexing range", "last", last, "indexed", indexed)
-			return 0
+			log.Warn("State indexer is in recovery", "indexed", indexed, "last", last)
+			return indexed - last
```

Comment:
should we return as `int64`, negative stands for the recovery stage

Reply:
I think it's OK, the number just indicates how many tasks to be done, uint64 will be simpler